src/charm.py

```diff
@@ -1571,6 +1576,13 @@ def _on_start(self, event: StartEvent) -> None:
             event.defer()
             return
 
+        if not self.get_secret(APP_SCOPE, "internal-ca"):
+            logger.info("leader not elected and/or internal CA not yet generated")
+            event.defer()
+            return
+        if not self.get_secret(UNIT_SCOPE, "internal-cert"):
+            self.tls.generate_internal_peer_cert()
+
```

Comment:
If the leader has generated the internal CA, the followers should generate their own certs.